src/main/java/com/example/datavisualizationbackend/visualizer/services/EventStorageService.java

```diff
@@ -0,0 +1,87 @@
+package com.example.datavisualizationbackend.visualizer.services;
+
+import com.example.datavisualizationbackend.visualizer.models.StoredEvent;
+import io.searchbox.client.JestClient;
+import io.searchbox.client.JestResult;
+import io.searchbox.core.Bulk;
+import io.searchbox.core.DeleteByQuery;
+import io.searchbox.core.Index;
+import io.searchbox.core.Search;
+import io.searchbox.indices.CreateIndex;
+import io.searchbox.indices.IndicesExists;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.util.List;
+
+@Service
+public class EventStorageService {
+
+    @Autowired
+    JestClient jestClient;
+
+    private String _indexName = "events";
+    private String _typeName = "event";
+
+    private static final Logger logger = LoggerFactory.getLogger(EventStorageService.class);
+
+    public void indexEvent(StoredEvent event) {
+
+        try {
+            IndicesExists indicesExists = new IndicesExists.Builder(_indexName).build();
+            JestResult result = jestClient.execute(indicesExists);
+
+            if (!result.isSucceeded()) {
+                CreateIndex createIndex = new CreateIndex.Builder(_indexName).build();
+                jestClient.execute(createIndex);
+            }
+
+            Bulk bulk = new Bulk.Builder()
+                    .addAction(new Index.Builder(event).index(_indexName).type(_typeName).build())
+                    .build();
+
+            result = jestClient.execute(bulk);
+
+            System.out.println(result.getJsonString());
+
+        } catch (IOException e) {
+            logger.error("Indexing error", e);
+        } catch (Exception e) {
+            logger.error("Indexing error", e);
+        }
+    }
+
+    public List<StoredEvent> getAllUploadEvents() {
+        try {
+            SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+            searchSourceBuilder.size(10000);
+            searchSourceBuilder.query(QueryBuilders.matchAllQuery());
+
+            Search search = new Search.Builder(searchSourceBuilder.toString())
+                    .addIndex(_indexName)
+                    .addType(_typeName)
+                    .build();
+
+            JestResult result = jestClient.execute(search);
+
+            return result.getSourceAsObjectList(StoredEvent.class);
+
+        } catch (IOException e) {
+            logger.error("Search error", e);
+        } catch (Exception e) {
```

Comment:
It's not usually a good idea to catch any possible exception. Just let the exception propagate up. Spring will catch and log it and your client will get a 500 error.